conan/tools/b2/b2deps.py

```diff
@@ -0,0 +1,284 @@
+            # Similarly declare the component targets, if any.
+            for name, component in dependency.cpp_info.get_sorted_components().items():
+                # Again, always skipping the kludge component as it's already
+                # defined.
+                if "_"+dep_name_b2 == name.lower():
+                    continue
+                cbiv += self._content_conanbuildinfo_variation_declare_target(
+                    dep_name_b2, name.lower(), component, settings=dep_settings, options=dep_options)
+            # Add the combined text.
+            self._content[dep_variant_jam] += "\n".join(cbiv)+"\n"
+
+    def _content_conanbuildinfo_variation_declare_libs(self, name, cpp_info, settings=None, options=None):
+        name = name.lower()
+        cbi_libs = []
+        variation = ' '.join(b2_features(b2_variation(settings, options)))
+        for lib in cpp_info.libs:
+            search = ' '.join(
+                [f'<search>"{b2_path(d.replace(self._conanhome, "$(CONAN_HOME)"))}"' for d in cpp_info.libdirs+cpp_info.bindirs])
+            # The lib targets are prefixed with "lib." to distinguish them
+            # from dependency main targets as it's often the case that the
+            # dependency has the same name as the library consumers link to.
+            cbi_libs += [
+                f'pkg-lib {name}//lib.{lib} : : <name>{lib}',
+                f'  {variation}',
+                f'  {search} ;']
+        return cbi_libs
+
+    def _content_conanbuildinfo_variation_declare_syslibs(self, name, systemlibs, settings=None, options=None):
+        name = name.lower()
+        cbi_libs = []
+        variation = ' '.join(b2_features(b2_variation(settings, options)))
+        for lib in systemlibs:
+            # Although system libs won't collide in the names. We still prefix
+            # the target names with "lib." for consistency and easier reference
+            # in the main targets.
+            cbi_libs += [
+                f'pkg-lib {name}//lib.{lib} : : <name>{lib}',
+                f'  {variation} ;']
+        return cbi_libs
+
+    def _content_conanbuildinfo_variation_declare_target(self, name, target, cpp_info, settings=None, options=None):
+        cbi_target = []
+        # Target, no sources. The empty target is to catch incompatible build
+        # requirements matches by falling back to an unbuildable result.
+        cbi_target += [
+            f'pkg-alias {name}//{target} : : <build>no ;',
+            f'pkg-alias {name}//{target} : :']
+        # Requirements:
+        cbi_target += [
+            f'  {" ".join(b2_features(b2_variation(settings, options)))}']
+        cbi_target += [
+            f'  <source>lib.{l}' for l in cpp_info.libs+cpp_info.system_libs]
+        # No default-build:
+        cbi_target += ["  : :"]
+        # Usage-requirements:
+        cbi_target += [
+            f'  <include>"{b2_path(d.replace(self._conanhome, "$(CONAN_HOME)"))}"' for d in cpp_info.includedirs]
```

Comment:
Is CONAN_HOME an environment variable? Because it will not be defined by Conan in Conan 2.0. Who is defining CONAN_HOME?